### PR TITLE
Godot patches

### PR DESCRIPTION
**Pull request by fire**
Here are the changes I did. Pushing for cicd testing.

### CODE REVIEW

File: test/data/Csaszar.ply [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
-3    0     6     4
-3    3     4     6
-3    1     2     3
-3    1     4     2
-3    1     0     4
-3    1     5     0
-3    3     5     4
-3    0     5     3
-3    0     3     2
+ply
```
Comment (fire) [inline test/data/Csaszar.ply:1]: These changes are probably not necessary. The test source changes aren't either.

File: test/polygon_test.cpp [hunk 2]
```diff
@@ -60,8 +61,8 @@ Polygons Turn180(Polygons polys) {
 }
 
 Polygons Duplicate(Polygons polys) {
-  float xMin = 1.0 / 0.0;
-  float xMax = -1.0 / 0.0;
+    float xMin = std::numeric_limits<float>::infinity();
```
Comment (fire) [inline test/polygon_test.cpp:64]: Whitespace.

File: tools/perf_test.cpp [hunk 3]
```diff
@@ -20,7 +20,7 @@
 using namespace manifold;
 
 int main(int argc, char **argv) {
-  for (int i = 0; i < 8; ++i) {
+  for (int i = 0; i < 3; ++i) {
```
Comment (fire) [inline tools/perf_test.cpp:23]: ?

**Comment by fire** [comment]
@elalish `1 workflow awaiting approval` Can you approve the workflow?

**Review by elalish — CHANGES_REQUESTED** [review]

**Comment by elalish** [comment]
CUDA compilation work on my machine, so i may just need to update the version of CUDA on the CI.

File: manifold/src/impl.cu [hunk 4]
```diff
@@ -414,22 +413,21 @@ int Manifold::Impl::InitializeNewReference(
...
+    graph.add_edge(edge.first, edge.second);
   }
-  std::vector<int> components(NumTri());
-  const int numComponent =
-      boost::connected_components(graph, components.data());
-
-  std::vector<int> comp2tri(numComponent, -1);
+  Components components = ConnectedComponents(graph);
+  const int numNodes = components.nodeComponents.size();
+  std::vector<int> comp2tri(numNodes, -1);
```
[the remark this answers is not in the document]
Comment (fire) [reply manifold/src/impl.cu:427]: I'm not sure how this worked at all.

[the remark this answers is not in the document]
Comment (fire) [reply manifold/src/impl.cu:427]: I think it worked by creating a component array that is semi-empty except for the components. 

A component has an index (unused) in the array and also the value of its root.

[the remark this answers is not in the document]
Comment (fire) [reply manifold/src/impl.cu:427]: Expected to have been given a list of { root tri, a vector of neighbours }.

**Comment by fire** [comment]
Rebased. 

Needs cicd approval.

Also, do geometry tests run on github actions?

**Comment by elalish** [comment]
Is this needed anymore now that #92 is merged?

**Comment by fire** [comment]
Let me check.

**Comment by fire** [comment]
@elalish Cicd approval required.

File: manifold/include/manifold.h [hunk 5]
```diff
@@ -16,7 +16,7 @@
 #include <functional>
 #include <memory>
 
-#include "structs.h"
+#include "../../utilities/include/structs.h"
```
[the remark this answers is not in the document]
Comment (fire) [reply manifold/include/manifold.h]: Not necessary.

**Review by elalish — APPROVED** [review]

**Pull request merged**